packages/paste-core/components/slider/src/Slider.tsx

```diff
@@ -0,0 +1,142 @@
+  const [hovered, setHovered] = React.useState(false);
+  const [focused, setFocused] = React.useState(false);
+
+  /*
+   * We deconstruct the props here to get the values we need
+   * and to preserve the underlying props object variable
+   */
+  const {
+    element = 'SLIDER',
+    minValue = 0,
+    maxValue = 100,
+    hideRangeLabels = false,
+    disabled: isDisabled = false,
+    id,
+  } = props;
+
+  // Our API differs from the underlying library, so we need to remap props
+  const remappedProps = {
+    ...props,
+    id,
+    isDisabled,
+    // needed to silence react-aria a11y guardrails
+    'aria-labelledby': id,
+  };
+
+  // These hooks manage the state of the slider
+  const state = useSliderState(remappedProps);
+  const {trackProps} = useSlider(remappedProps, state, trackRef);
+  const {thumbProps, inputProps, isDragging} = useSliderThumb(
+    {
+      index: 0,
+      trackRef,
+      inputRef,
+    },
+    state
+  );
+
+  // Used to determine the UI styling of the track and thumb
+  const uiStateProps = React.useMemo(() => {
+    return {
+      disabled: isDisabled,
+      error: props.hasError,
+      hovered,
+      focused,
+      dragging: isDragging,
+    };
+  }, [isDisabled, props.hasError, hovered, focused, isDragging]);
+
+  return (
+    <Box role="group" element={element}>
+      {/* Create a container for the optional min and max values */}
+      {!hideRangeLabels && (
+        <Box
+          role="presentation"
+          element={`${element}_RANGE_LABELS`}
+          display="flex"
+          justifyContent="space-between"
+          fontSize="fontSize30"
+          lineHeight="lineHeight30"
+          fontWeight="fontWeightSemibold"
+          color="colorTextWeak"
+        >
+          <Box as="span" element={`${element}_RANGE_LABELS_MIN`}>
+            {props.numberFormatter.format(minValue)}
+          </Box>
+          <Box as="span" element={`${element}_RANGE_LABELS_MAX`}>
+            {props.numberFormatter.format(maxValue)}
+          </Box>
+        </Box>
+      )}
+      {/* The track element holds the visible track line and the thumb */}
+      <Box
+        {...trackProps}
+        element={`${element}_TRACK_CONTAINER`}
+        ref={trackRef}
+        onMouseEnter={() => setHovered(true)}
+        onMouseLeave={() => setHovered(false)}
+        display="flex"
+        alignItems="center"
+        height="20px"
+        width="100%"
+        cursor="pointer"
+      >
+        <StyledTrack
+          {...uiStateProps}
+          element={`${element}_TRACK`}
+          left={thumbProps.style?.left}
+          height="4px"
+          width="100%"
+          borderRadius="borderRadius20"
+          cursor="pointer"
+        >
+          <SliderThumb {...thumbProps} {...uiStateProps} element={`${element}_THUMB`}>
+            <ScreenReaderOnly>
+              <input
+                ref={inputRef}
+                {...inputProps}
+                aria-labelledby={undefined}
```

Comment:
issue: this also needs to be described by the help text. It might work because of the props being spread but our examples should show and implement this

Reply:
Fixed in the docs PR, but brought it over to this PR. You are 100% correct. Great catch

Reply:
Input storybook doesn't implement this well, but website examples do. Fixing storybook examples for Input.